### PR TITLE
feat: add admin.hidden to collections and globals

### PR DESCRIPTION
**Pull request by DanRibbens**
## Description

Allows configuation of collections and globals to be excluded from the admin UI in navigation and routing using `admin.hidden`.

- [x] I have read and understand the CONTRIBUTING.md document in this repository

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Existing test suite passes locally with my changes
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by DanRibbens** [comment]
As I think about this more, `admin.hidden` should be a function that gets called with the user object so that it can be dynamic.

File: src/admin/components/Routes.tsx
Comment (DanRibbens) [inline src/admin/components/Routes.tsx:1]: GitHub shit the bed on this diff. The only line really changed here is the `.filter()` added at L180 and the rest is spacing.

File: src/admin/components/views/Dashboard/Default.tsx [hunk 1]
```diff
@@ -41,24 +42,28 @@ const Dashboard: React.FC<Props> = (props) => {
 
   useEffect(() => {
     setGroups(groupNavItems([
-      ...collections.map((collection) => {
-        const entityToGroup: EntityToGroup = {
-          type: EntityType.collection,
-          entity: collection,
-        };
+      ...collections
+        .filter(({ admin: { hidden } }) => !(typeof hidden === 'function' ? hidden({ user }) : hidden))
```
Comment (denolfe) [inline src/admin/components/views/Dashboard/Default.tsx:46]: Any reason this isn't a `!==` ?

Comment (DanRibbens) [reply src/admin/components/views/Dashboard/Default.tsx:46]: Not sure I follow... `hidden` could be undefined, a function or a boolean. I'm not sure I can make this more succinct.

**Comment by TimHal** [comment]
Thanks a lot for this! It almost went unnoticed for me, but this pretty much solves https://github.com/payloadcms/payload/discussions/1462

Should I close the discussion and link this PR as solution?

**Comment by philipgher** [comment]
This feature is exactly what I was in need for just now. Just upgraded and tested it out. I noticed I am not getting the full user object however which makes it unusable for making the collection hidden based on the role. I am getting this data atm:
```
user: {
  collection: "users"
  email: "email@test.com"
  exp: 1683014438
  iat: 1683007238
  id: "644d5bd68438c28453441aaf"
}
```

Is this the intended return object or is this a bug @DanRibbens ?

**Comment by DanRibbens** [comment]
That user object from the decoded JWT token from the auth provider component in the admin UI.

To add more properties to this object you can add `saveToJWT: true` to any fields in your user collection.

I do agree it would be nicer to have the full user object passed through `useAuth`. Changing this would be a breaking change since some of the properties would not be in place anymore. If this is a change you want, I would start a new discussion so it isn't lost in this old PR thread.

**Comment by DanRibbens** [comment]
> Thanks a lot for this! It almost went unnoticed for me, but this pretty much solves #1462
> 
> Should I close the discussion and link this PR as solution?

@TimHal That would be great!

**Comment by philipgher** [comment]
Ahh right that makes sense once you know it. Thanks for explaining this @DanRibbens. By reading through the changes in this PR I did notice the data indirectly came from the `useAuth` hook so I read the docs on that hook. In there it is not mentioned the user object is actually not the full data but only the things stored in the token.

Maybe a documentation update would suffice here. Maybe getting the full user object would be nice if there's no (acceptable) performance hit or something like that.

https://payloadcms.com/docs/admin/hooks#useauth

**Pull request merged**